cpg_workflows/stages/large_cohort.py

```diff
@@ -12,73 +14,110 @@
     get_workflow,
     stage,
 )
+from metamist.graphql import gql, query
 
-from .genotype import Genotype
+if TYPE_CHECKING:
+    from graphql import DocumentNode
 
+    from hailtop.batch.job import PythonJob
 
-@stage(required_stages=[Genotype])
+
+@stage(analysis_type="combiner", analysis_keys=["combiner"])
 class Combiner(CohortStage):
     def expected_outputs(self, cohort: Cohort) -> Path:
-        if vds_version := get_config()['workflow'].get('vds_version'):
-            vds_version = slugify(vds_version)
-            if not vds_version.startswith('v'):
-                vds_version = f'v{vds_version}'
-
-        vds_version = vds_version or get_workflow().output_version
-        return cohort.analysis_dataset.prefix() / 'vds' / f'{vds_version}.vds'
+        workflow_config = config_retrieve("workflow")
+        combiner_config = config_retrieve("combiner")
+        output_vds_name: str = slugify(
+            f"{workflow_config['cohort']}-{workflow_config['sequencing_type']}-{combiner_config['vds_version']}",
+        )
+        return cohort.analysis_dataset.prefix() / "vds" / f"{output_vds_name}.vds"
+
+    def get_vds_ids_output(self, vds_id: int) -> list[str]:
+        get_vds_analysis_query: DocumentNode = gql(
```

Comment:
If you create this at the root of the script (as with [this](https://github.com/populationgenomics/production-pipelines/blob/d5bf2b8ae33b5274acf16915ebde1075302a2316/cpg_workflows/metamist.py#L34)) you can do GraphQL validation in unit tests. Not that we do that, but you can.